README.md

```diff
@@ -1,34 +1,93 @@
-# Membrane Multimedia Framework: Template Plugin
+# Membrane FFmpeg SWScale plugin
 
-[![Hex.pm](https://img.shields.io/hexpm/v/membrane_template_plugin.svg)](https://hex.pm/packages/membrane_template_plugin)
-[![CircleCI](https://circleci.com/gh/membraneframework/membrane_template_plugin.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_template_plugin)
+[![Hex.pm](https://img.shields.io/hexpm/v/membrane_ffmpeg_swscale_plugin.svg)](https://hex.pm/packages/membrane_ffmpeg_swscale_plugin)
+[![API Docs](https://img.shields.io/badge/api-docs-yellow.svg?style=flat)](https://hexdocs.pm/membrane_ffmpeg_swscale_plugin/)
+[![CircleCI](https://circleci.com/gh/membraneframework/membrane_ffmpeg_swscale_plugin.svg?style=svg)](https://circleci.com/gh/membraneframework/membrane_ffmpeg_swscale_plugin)
 
-This repository contains a template for new plugins.
+Plugin performing video scaling, using SWScale module of [FFmpeg](https://www.ffmpeg.org/) library.
 
-Check out different branches for other flavours of template.
-
-It is part of [Membrane Multimedia Framework](https://membraneframework.org).
+It is a part of [Membrane Multimedia Framework](https://membraneframework.org).
 
 ## Installation
 
-The package can be installed by adding `membrane_template_plugin` to your list of dependencies in `mix.exs`:
+Add the following line to your `deps` in `mix.exs`. Run `mix deps.get`.
 
 ```elixir
-def deps do
-  [
-    {:membrane_template_plugin, "~> 0.1.0"}
-  ]
+{:membrane_ffmpeg_swscale_plugin, "~> 0.1.0"}
+```
+
+You also need to have [FFmpeg](https://www.ffmpeg.org/) library installed.
+
+## Sample usage
+Scaler needs input in the YUV420p format. It processes one frame at a time. The element also requires
+getting caps with video width and height. To meet all requirements either `Membrane.Element.RawVideo.Parser` or `Membrane.H264.FFmpeg.Decoder`
+can be used before Scaler in the pipeline.
+
+### Raw video format
+If you want to scale a video in YUV420p format, you need to parse it firstly like in the example below:
+```elixir
+defmodule Scaling.Pipeline do
+  use Membrane.Pipeline
+
+  @doc false
+  @impl true
+  def handle_init(_) do
+    children = [
+      file_src: %Membrane.File.Source{location: "/tmp/input.raw"},
+      parser: %Membrane.Element.RawVideo.Parser{format: :I420, width: 1280, height: 720},
+      scaler: %Membrane.FFmpeg.SWScale.Scaler{output_width: 640, output_height: 640},
+      file_sink: %Membrane.File.Sink{location: "/tmp/output.raw"}
+    ]
+
+    links = [
+      link(:file_src)
+      |> to(:parser)
+      |> to(:scaler)
+      |> to(:file_sink)
+    ]
+
+    {{:ok, spec: %ParentSpec{children: children, links: links}}, %{}}
+  end
 end
 ```
+Result file will contain video also in YUV420p format.
+
+### H.264
+Scaling encoded (using H.264 standard) video requires parser and decoder because Scaler scales only raw video.
```

Comment:
I'm not sure if 2 examples are needed, they are very similar, I'd only left this one.